lib/Field/FieldInStructureEditModeProvider.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { node, bool } from 'prop-types';
+
+export const FieldInStructureEditModeContext = React.createContext({});
+
+export const FieldInStructureEditModeProvider = ({
```

Comment:
🔴 After re-reading the PR a few times I'm still a little confused by the need for this context.

We've not got the ability to pass in props directly to the child components to say if they are `editable` or not so do we need this anymore?

---

The only argument I can see is the need to set this once and only once. So let's dig a little deeper;

There is currently three occurrences of this context. Twice for showing form elements and once for deciding what layout class we render.

For the layout class, I have added some separate feedback (basically we don't need to do this). 

(IMO) The other 2 instances don't support the need to have context. It's actually quite nice to pass the editable prop directly to the child for readability.

```
// This label is editable cause I can see the prop - no need for a new abstraction
<FieldLabel editable />
```

It's also fixed, which is quite strange. If it was sharing state that change based on user interaction then that would make more sense because we're abstracting logic but here we're just sharing a prop...with composable components.

-----

Let me know if you need to pair on this for clarification.

Reply:
After chatting I can see the benefits of this!

If possible let's utilise the base `Field` component and share some state (using context) in there rather than having this abstraction.